components/nexus-mime/src/main/java/org/sonatype/nexus/mime/internal/DefaultMimeSupport.java

```diff
@@ -97,49 +99,70 @@ public DefaultMimeSupport(final NexusMimeTypes nexusMimeTypes) {
             metadata.set(Metadata.RESOURCE_NAME_KEY, "dummy." + key);
             MediaType mediaType = detector.detect(null, metadata);
             // unravel to least specific
-            while (mediaType != null) {
-              detected.add(mediaType.getBaseType().toString());
-              mediaType = tikaConfig.getMediaTypeRegistry().getSupertype(mediaType);
-            }
+            unravel(detected, mediaType);
             return detected;
           }
         });
   }
 
+  @Nonnull
   @Override
-  public String guessMimeTypeFromPath(final MimeRulesSource mimeRulesSource, final String path) {
+  public String guessMimeTypeFromPath(@Nullable final MimeRulesSource mimeRulesSource, final String path) {
+    return guessMimeTypesListFromPath(mimeRulesSource, path).get(0);
+  }
+
+  @Nonnull
+  @Override
+  public List<String> guessMimeTypesListFromPath(@Nullable final MimeRulesSource mimeRulesSource, final String path) {
+    checkNotNull(path);
+    List<String> mimeTypes = new ArrayList<>();
     if (mimeRulesSource != null) {
-      final String hardRule = mimeRulesSource.getRuleForPath(path);
-      if (!Strings.isNullOrEmpty(hardRule)) {
-        return hardRule;
+      final MimeRule hardRule = mimeRulesSource.getRuleForName(path);
```

Comment:
It is always the extension that is used only, so basically "file path", "file name" does not matter, it's getting the extension that matters. Maybe rename to "name" only?